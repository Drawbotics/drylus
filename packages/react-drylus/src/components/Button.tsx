import sv, { fade } from '@drawbotics/drylus-style-vars';
import { css, cx } from 'emotion';
import React from 'react';

import { Category, Color, Size, Tier } from '../enums';
import { OnClickCallback, Responsive, Style } from '../types';
import { colorEnumToCategory, getEnumAsClass, run, useResponsiveProps } from '../utils';

const styles = {
  root: css`
    background: ${sv.neutralLight};
    color: ${sv.colorPrimary};
    border-radius: ${sv.defaultBorderRadius};
    padding: calc(${sv.paddingExtraSmall} * 1.5) ${sv.defaultPadding};
    outline: 0;
    border: 0;
    transition: ${sv.transitionShort};
    display: inline-flex;
    align-items: center;
    justify-content: center;

    &:hover {
      cursor: pointer;
      background: ${sv.neutral};
    }

    &:active {
      box-shadow: ${sv.insetActiveMedium};
    }

    &:disabled {
      cursor: not-allowed;
      background: ${sv.neutralLight};
      color: ${sv.colorDisabled};
      box-shadow: none;
    }
  `,
  brand: css`
    background: ${sv.brand};
    color: ${sv.white} !important;

    &:hover {
      background: ${sv.brandDark};
    }

    &:active {
      box-shadow: ${sv.insetActive};
    }

    &:disabled {
      background: ${fade(sv.brand, 40)};
    }
  `,
  danger: css`
    background: ${sv.red};
    color: ${sv.white} !important;

    &:hover {
      background: ${sv.redDark};
    }

    &:active {
      box-shadow: ${sv.insetActive};
    }

    &:disabled {
      background: ${fade(sv.red, 40)};
    }
  `,
  info: css`
    background: ${sv.blue};
    color: ${sv.white} !important;

    &:hover {
      background: ${sv.blueDark};
    }

    &:active {
      box-shadow: ${sv.insetActive};
    }

    &:disabled {
      background: ${fade(sv.blue, 40)};
    }
  `,
  success: css`
    background: ${sv.green};
    color: ${sv.white} !important;

    &:hover {
      background: ${sv.greenDark};
    }

    &:active {
      box-shadow: ${sv.insetActive};
    }

    &:disabled {
      background: ${fade(sv.green, 40)};
    }
  `,
  warning: css`
    background: ${sv.orange};
    color: ${sv.white} !important;

    &:hover {
      background: ${sv.orangeDark};
    }

    &:active {
      box-shadow: ${sv.insetActive};
    }

    &:disabled {
      background: ${fade(sv.orange, 40)};
    }
  `,
  brandAlt: css`
    color: ${sv.brand};

    &:hover {
      color: ${sv.brandDark};
    }

    &:disabled {
      color: ${fade(sv.brand, 40)};
    }
  `,
  dangerAlt: css`
    color: ${sv.red};

    &:hover {
      color: ${sv.redDark};
    }

    &:disabled {
      color: ${fade(sv.red, 40)};
    }
  `,
  warningAlt: css`
    color: ${sv.orange};

    &:hover {
      color: ${sv.orangeDark};
    }

    &:disabled {
      color: ${fade(sv.orange, 40)};
    }
  `,
  infoAlt: css`
    color: ${sv.blue};

    &:hover {
      color: ${sv.blueDark};
    }

    &:disabled {
      color: ${fade(sv.blue, 40)};
    }
  `,
  successAlt: css`
    color: ${sv.green};

    &:hover {
      color: ${sv.greenDark};
    }

    &:disabled {
      color: ${fade(sv.green, 40)};
    }
  `,
  primaryAlt: css`
    color: ${sv.colorPrimary};

    &:hover {
      color: ${sv.colorPrimary};
    }

    &:disabled {
      color: ${sv.colorDisabled};
    }
  `,
  small: css`
    padding: calc(${sv.paddingExtraSmall} - 1px) ${sv.paddingExtraSmall};
    /* font-size: 0.9rem; */
  `,
  large: css`
    padding: ${sv.paddingSmall} ${sv.paddingHuge};
  `,
  secondary: css`
    background: transparent;
    color: ${sv.colorPrimary};
    box-shadow: 0 0 0 1px ${sv.neutral} inset;

    &:hover {
      background: ${sv.neutralLight};
    }

    &:active {
      box-shadow: 0 0 0 1px ${sv.neutral} inset, ${sv.insetActiveLight};
    }

    &:disabled {
      box-shadow: 0 0 0 1px ${sv.neutralLight} inset;
      background: none;
    }
  `,
  tertiary: css`
    background: transparent;
    color: ${sv.blue};

    &:hover {
      background: ${fade(sv.neutral, 50)};
      color: ${sv.blueDark};
    }

    &:active {
      box-shadow: ${sv.insetActiveLight};
    }

    &:disabled {
      background: none;
      color: ${sv.colorDisabled};
    }
  `,
  round: css`
    border-radius: 1000px;
    height: ${sv.marginExtraLarge};
    width: ${sv.marginExtraLarge};
    padding: 0;

    i {
      display: flex;
      align-items: center;
      justify-content: center;
      margin-right: 0;
      margin-left: -1px;
      margin-bottom: -1px;
    }
  `,
  roundSmall: css`
    height: ${sv.marginLarge};
    width: ${sv.marginLarge};

    i {
      font-size: 1rem;
      margin-bottom: -1px;
      margin-left: -1px;
    }
  `,
  fullWidth: css`
    width: 100%;
    flex: 1;
  `,
  leading: css`
    margin-right: ${sv.marginExtraSmall};
    height: 0;
    display: flex;
    align-items: center;
  `,
  trailing: css`
    margin-left: ${sv.marginExtraSmall};
    height: 0;
    display: flex;
    align-items: center;
  `,
};

export const buttonStyles = styles;

export interface ButtonProps {
  children?: React.ReactNode;

  /** Disables the button click */
  disabled?: boolean;

  /** Triggered after the button is clicked */
  onClick?: OnClickCallback<HTMLElement>;

  category?: Exclude<Category, Category.PRIMARY>;

  color?: Exclude<Color, Color.PRIMARY>;

  /** @default Size.DEFAULT */
  size?: Extract<Size, Size.SMALL | Size.DEFAULT | Size.LARGE>;

  /** @default Tier.PRIMARY */
  tier?: Tier;

  /** Shown in front of the button text, can be a Spinner or Icon */
  leading?: React.ReactNode;

  /** Shown after the button text, can be a Spinner or Icon */
  trailing?: React.ReactNode;

  /** Makes button take the full width of the container */
  fullWidth?: boolean;

  /** Used for style overrides */
  style?: Style;

  /** Reponsive prop overrides */
  responsive?: Responsive<this>;
}

export const Button = ({ responsive, ...rest }: ButtonProps) => {
  const {
    children,
    disabled,
    onClick,
    category: _category,
    size = Size.DEFAULT,
    tier = Tier.PRIMARY,
    leading,
    trailing,
    fullWidth,
    style,
    color,
  } = useResponsiveProps<ButtonProps>(rest, responsive);

  if (!children && trailing && leading) {
    throw new Error('If no children are given, only pass trailing or leading, but not both');
  }

  const round = children == null && (trailing != null || leading != null);

  const category = color ? colorEnumToCategory(color) : _category;

  return (
    <button
      style={style}
      onClick={onClick}
      className={cx(styles.root, {
        [styles[getEnumAsClass<typeof styles>(size)]]: size != null,
        [styles.round]: round === true,
        [styles.roundSmall]: round === true && size === Size.SMALL,
        [styles[getEnumAsClass<typeof styles>(category)]]:
          category != null && tier === Tier.PRIMARY,
        [styles[getEnumAsClass<typeof styles>(tier)]]: tier != null,
        [styles[`${category?.toLowerCase() ?? ''}Alt` as keyof typeof styles]]:
          category != null && tier !== Tier.PRIMARY,
        [styles.fullWidth]: fullWidth === true,
      })}
      disabled={disabled}>
      {run(() => {
        if (leading) {
          return <div className={cx({ [styles.leading]: !round })}>{leading}</div>;
        }
      })}
      {children}
      {run(() => {
        if (trailing) {
          return <div className={cx({ [styles.trailing]: !round })}>{trailing}</div>;
        }
      })}
    </button>
  );
};
