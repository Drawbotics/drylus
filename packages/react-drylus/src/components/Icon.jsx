import React from 'react';
import { css, cx, injectGlobal } from 'emotion';
import PropTypes from 'prop-types';
import generateIconStyles from '@drawbotics/icons/dist/drycons.js';
import packageJson from '@drawbotics/icons/package.json';
import sv from '@drawbotics/drylus-style-vars';

import { Categories } from '../base';
import { getEnumAsClass } from '../utils';
import env from '../utils/get-static-env';

console.log(env);


injectGlobal`
  ${generateIconStyles(env === 'development' ? 'dev' : packageJson.version)}
`;


const styles = {
  root: css`
    color: inherit;
  `,
  bold: css`
    font-weight: bold !important;
  `,
  danger: css`
    color: ${sv.red};
  `,
  info: css`
    color: ${sv.blue};
  `,
  success: css`
    color: ${sv.green};
  `,
  warning: css`
    color: ${sv.orange};
  `,
  brand: css`
    color: ${sv.brand};
  `,
  clickable: css`
    &:hover {
      cursor: pointer;
    }
  `,
};


const Icon = ({ name, bold, onClick, category }) => {
  return (
    <i className={cx(styles.root, `Drycon Drycon-${name}`, {
      [styles.bold]: bold,
      [styles[getEnumAsClass(category)]]: category,
      [styles.clickable]: onClick,
    })} onClick={onClick} />
  );
}


Icon.propTypes = {
  /** Name of the icon */
  name: PropTypes.string.isRequired,

  /** Makes icon T H I C C */
  bold: PropTypes.bool,

  /** Triggered when the icon is clicked */
  onClick: PropTypes.func,

  /** Category of the icon */
  category: PropTypes.oneOf([
    Categories.DANGER,
    Categories.INFO,
    Categories.SUCCESS,
    Categories.WARNING,
    Categories.BRAND,
  ]),
}


export default Icon;
