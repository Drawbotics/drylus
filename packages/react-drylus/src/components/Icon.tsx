import sv from '@drawbotics/drylus-style-vars';
import { IconKeys, Icons as IconValues } from '@drawbotics/icons/dist/drycons';
import { generateIconStyles, mapping } from '@drawbotics/icons/dist/drycons.js';
import packageJson from '@drawbotics/icons/package.json';
import { css, cx, injectGlobal } from 'emotion';
import React from 'react';

import { Category, Color, Shade } from '../enums';
import { OnClickCallback, Style } from '../types';
import { Deprecated, categoryEnumToColor, getEnumAsClass } from '../utils';

const env = require('../utils/get-static-env');

injectGlobal`
  ${generateIconStyles(env === "'development'" ? 'dev' : packageJson.version)}
`;

const styles = {
  root: css`
    color: inherit;
  `,
  bold: css`
    font-weight: bold !important;
  `,
  red: css`
    color: ${sv.red};
  `,
  blue: css`
    color: ${sv.blue};
  `,
  green: css`
    color: ${sv.green};
  `,
  orange: css`
    color: ${sv.orange};
  `,
  brand: css`
    color: ${sv.brand};
  `,
  primary: css`
    color: ${sv.colorPrimary};
  `,
  light: css`
    color: ${sv.colorTertiary};
  `,
  medium: css`
    color: ${sv.colorSecondary};
  `,
  dark: css`
    color: ${sv.colorPrimary};
  `,
  clickable: css`
    &:hover {
      cursor: pointer;
    }
  `,
};

export const Icons: Record<IconKeys, IconValues> = mapping;

export type IconType = IconValues;

export interface IconProps {
  /** Name of the icon */
  name: IconType;

  /** Makes icon bold */
  bold?: boolean;

  /** Triggered when the icon is clicked */
  onClick?: OnClickCallback<HTMLElement>;

  /** @deprecated Use color instead */
  category?: Category;

  color?: Color;

  /** @kind Shade */
  shade?: Shade.DARK | Shade.MEDIUM | Shade.LIGHT;

  /** Used for style overrides */
  style?: Style;
}

export const Icon = ({ name, bold, onClick, category, style, color: _color, shade }: IconProps) => {
  const color = category ? categoryEnumToColor(category) : _color;
  return (
    <i
      style={style}
      className={cx(styles.root, `Drycon Drycon-${name}`, {
        [styles.bold]: bold != null,
        [styles[getEnumAsClass<typeof styles>(color)]]: color != null,
        [styles[getEnumAsClass<typeof styles>(shade)]]: shade != null,
        [styles.clickable]: onClick != null,
      })}
      onClick={onClick}
    />
  );
};

Icon.propTypes = {
  category: Deprecated,
};
