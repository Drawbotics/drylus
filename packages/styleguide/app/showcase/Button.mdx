import { Button } from '@drawbotics/react-drylus';

import Playground from '../components/Playground';


# Hello

The following is an non-editable code example
```jsx
<Button type="danger">Hello</Button>
```

The following is the rendered component
<Playground component={Button}>
  <Button>Hello</Button>
  <Button>Fuck</Button>
</Playground>
