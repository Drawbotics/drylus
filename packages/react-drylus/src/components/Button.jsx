import React from 'react';
import { css, cx } from 'emotion';
import PropTypes from 'prop-types';
import sv from '@drawbotics/style-vars';

import { Categories, Sizes } from '../base';
import Icon from './Icon';


const styles = {
  root: css`
    background: ${sv.neutralLight};
    color: ${sv.colorPrimary};
    border-radius: ${sv.defaultBorderRadius};
    padding: calc(${sv.paddingExtraSmall} * 1.5) ${sv.defaultPadding};
    outline: 0;
    border: 0;
    transition: ${sv.defaultTransition};
    display: inline-flex;
    align-items: center;
    justify-content: center;

    &:hover {
      cursor: pointer;
      background: ${sv.neutral};
    }

    &:active {
      box-shadow: ${sv.insetActiveMedium};
    }

    &:disabled {
      cursor: not-allowed;
      background: ${sv.neutralLight} !important;
      color: ${sv.colorDisabled} !important;
      box-shadow: none;
    }
  `,
  brand: css`
    background: ${sv.brand};
    color: ${sv.white};

    &:hover {
      background: ${sv.brandDark};
    }

    &:active {
      box-shadow: ${sv.insetActive};
    }
  `,
  danger: css`
    background: ${sv.red};
    color: ${sv.white};

    &:hover {
      background: ${sv.redDark};
    }

    &:active {
      box-shadow: ${sv.insetActive};
    }
  `,
  info: css`
    background: ${sv.blue};
    color: ${sv.white};

    &:hover {
      background: ${sv.blueDark};
    }

    &:active {
      box-shadow: ${sv.insetActive};
    }
  `,
  success: css`
    background: ${sv.green};
    color: ${sv.white};

    &:hover {
      background: ${sv.greenDark};
    }

    &:active {
      box-shadow: ${sv.insetActive};
    }
  `,
  warning: css`
    background: ${sv.orange};
    color: ${sv.white};

    &:hover {
      background: ${sv.orangeDark};
    }

    &:active {
      box-shadow: ${sv.insetActive};
    }
  `,
  small: css`
    padding: ${sv.paddingExtraSmall} ${sv.paddingExtraSmall};
    font-size: 0.9rem;
  `,
  large: css`
    padding: ${sv.paddingSmall} ${sv.paddingHuge};
  `,
  secondary: css`
    background: transparent;
    color: ${sv.colorPrimary};
    box-shadow: 0 0 0 1px ${sv.neutral} inset;

    &:hover {
      background: ${sv.neutralLight};
    }

    &:active {
      box-shadow: ${sv.insetActiveLight};
    }
  `,
  tertiary: css`
    background: transparent;
    color: ${sv.blue};

    &:hover {
      background: ${sv.neutralLight};
      color: ${sv.blueDark};
    }

    &:active {
      box-shadow: ${sv.insetActiveLight};
    }
  `,
  rightIcon: css`
    > i {
      margin-top: -2px;
      margin-left: 4px;
    }
  `,
  leftIcon: css`
    > i {
      margin-top: -2px;
      margin-right: 4px;
    }
  `,
  iconOnly: css`
    border-radius: 1000px;
    height: ${sv.marginExtraLarge};
    width: ${sv.marginExtraLarge};
    padding: 0;

    > i {
      display: flex;
      align-items: center;
      justify-content: center;
      margin-right: 0;
      margin-left: 0;
      margin-bottom: -3px;
    }
  `,
  smallIcon: css`
    height: ${sv.marginLarge};
    width: ${sv.marginLarge};

    > i {
      font-size: 1rem;
      margin-bottom: -3px;
    }
  `,
};


export const ButtonTiers = {
  PRIMARY: 'PRIMARY',
  SECONDARY: 'SECONDARY',
  TERTIARY: 'TERTIARY',
};


const Button = ({
  children,
  disabled,
  onClick,
  category,
  size,
  tier=ButtonTiers.PRIMARY,
  icon,
  iconSide='left',
}) => {
  const iconOnly = ! children && icon;
  return (
    <button
      onClick={onClick}
      className={cx(styles.root, {
        [styles[size?.toLowerCase()]]: size,
        [styles[tier?.toLowerCase()]]: tier,
        [styles.rightIcon]: iconSide === 'right' && icon,
        [styles.leftIcon]: iconSide === 'left' && icon,
        [styles.iconOnly]: iconOnly,
        [styles.smallIcon]: iconOnly && size === 'SMALL',
        [styles[category?.toLowerCase()]]: category && tier === ButtonTiers.PRIMARY,
      })}
      disabled={disabled}>
      {do{
        if (iconSide === 'left' && icon) {
          <Icon name={icon} />
        }
      }}
      {children}
      {do{
        if (iconSide === 'right' && icon) {
          <Icon name={icon} />
        }
      }}
    </button>
  );
}


Button.propTypes = {
  /** Just text for the button */
  children: PropTypes.string,

  /** Disables the button click */
  disabled: PropTypes.bool,

  /** Triggered after the button is clicked */
  onClick: PropTypes.func,

  /** Category of the button. Can be danger, success, info, warning */
  category: PropTypes.oneOf([Categories.BRAND, Categories.DANGER, Categories.SUCCESS, Categories.INFO, Categories.WARNING]),

  /** Size of the button. Can be small, large, default */
  size: PropTypes.oneOf([Sizes.SMALL, Sizes.DEFAULT, Sizes.LARGE]),

  /** Tier of the button */
  tier: PropTypes.oneOf([ButtonTiers.PRIMARY, ButtonTiers.SECONDARY, ButtonTiers.TERTIARY]),

  /** Name of the icon to be displayed within the button. Shown on the left by default */
  icon: PropTypes.string,

  /** Side on which the icon is displayed */
  iconSide: PropTypes.oneOf(['left', 'right']),
};


Button.defaultProps = {
  iconSide: 'left',
  size: Sizes.DEFAULT,
  tier: ButtonTiers.PRIMARY,
};


export default Button;
