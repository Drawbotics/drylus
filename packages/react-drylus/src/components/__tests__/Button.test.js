import React from 'react';
import { create } from 'react-test-renderer';

import Button from '../Button';
import {
  Size,
  Tier,
  Category,
} from '../../enums';
import Icon from '../Icon';


describe('Button', () => {
  describe('matches snapshot when', () => {
    it('is basic', () => {
      const tree = create(
        <Button>
          Some content
        </Button>
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('has a category', () => {
      const tree = create(
        <Button category={Category.WARNING}>
          Some content
        </Button>
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('has a tier', () => {
      const tree = create(
        <Button tier={Tier.SECONDARY}>
          Some content
        </Button>
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('is small', () => {
      const tree = create(
        <Button size={Size.SMALL}>
          Some content
        </Button>
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('has a leading component', () => {
      const tree = create(
        <Button leading={<Icon name="check" />}>
          Some content
        </Button>
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('has a trailing component', () => {
      const tree = create(
        <Button trailing={<Icon name="check" />}>
          Some content
        </Button>
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});