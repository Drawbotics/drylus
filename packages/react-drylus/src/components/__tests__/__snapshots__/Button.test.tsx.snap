// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches snapshot when has a category 1`] = `
.emotion-0 {
  background: #eaeff4;
  color: #172b4e;
  border-radius: 5px;
  padding: calc(8px * 1.5) 24px;
  outline: 0;
  border: 0;
  -webkit-transition: all 0.15s ease-in-out;
  transition: all 0.15s ease-in-out;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: #f8a00f;
  color: #fff !important;
}

.emotion-0:hover {
  cursor: pointer;
  background: #d3dce6;
}

.emotion-0:active {
  box-shadow: 0 1px 6px #879ba6 inset;
}

.emotion-0:disabled {
  cursor: not-allowed;
  background: #eaeff4;
  color: rgba(135,155,166,0.5);
  box-shadow: none;
}

.emotion-0:hover {
  background: #a16d22;
}

.emotion-0:active {
  box-shadow: 0 1px 6px #172b4e inset;
}

.emotion-0:disabled {
  background: rgba(248,160,15,0.4);
}

<button
  className="undefined emotion-0"
>
  Some content
</button>
`;

exports[`Button matches snapshot when has a leading component 1`] = `
.emotion-2 {
  background: #eaeff4;
  color: #172b4e;
  border-radius: 5px;
  padding: calc(8px * 1.5) 24px;
  outline: 0;
  border: 0;
  -webkit-transition: all 0.15s ease-in-out;
  transition: all 0.15s ease-in-out;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.emotion-2:hover {
  cursor: pointer;
  background: #d3dce6;
}

.emotion-2:active {
  box-shadow: 0 1px 6px #879ba6 inset;
}

.emotion-2:disabled {
  cursor: not-allowed;
  background: #eaeff4;
  color: rgba(135,155,166,0.5);
  box-shadow: none;
}

.emotion-1 {
  margin-right: 8px;
  height: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 {
  color: inherit;
}

<button
  className="emotion-2"
>
  <div
    className="emotion-1"
  >
    <i
      className="emotion-0 Drycon Drycon-check"
    />
  </div>
  Some content
</button>
`;

exports[`Button matches snapshot when has a tier 1`] = `
.emotion-0 {
  background: #eaeff4;
  color: #172b4e;
  border-radius: 5px;
  padding: calc(8px * 1.5) 24px;
  outline: 0;
  border: 0;
  -webkit-transition: all 0.15s ease-in-out;
  transition: all 0.15s ease-in-out;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: transparent;
  color: #172b4e;
  box-shadow: 0 0 0 1px #d3dce6 inset;
}

.emotion-0:hover {
  cursor: pointer;
  background: #d3dce6;
}

.emotion-0:active {
  box-shadow: 0 1px 6px #879ba6 inset;
}

.emotion-0:disabled {
  cursor: not-allowed;
  background: #eaeff4;
  color: rgba(135,155,166,0.5);
  box-shadow: none;
}

.emotion-0:hover {
  background: #eaeff4;
}

.emotion-0:active {
  box-shadow: 0 0 0 1px #d3dce6 inset,0 1px 6px #d3dce6 inset;
}

.emotion-0:disabled {
  box-shadow: 0 0 0 1px #eaeff4 inset;
  background: none;
}

<button
  className="emotion-0"
>
  Some content
</button>
`;

exports[`Button matches snapshot when has a trailing component 1`] = `
.emotion-2 {
  background: #eaeff4;
  color: #172b4e;
  border-radius: 5px;
  padding: calc(8px * 1.5) 24px;
  outline: 0;
  border: 0;
  -webkit-transition: all 0.15s ease-in-out;
  transition: all 0.15s ease-in-out;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.emotion-2:hover {
  cursor: pointer;
  background: #d3dce6;
}

.emotion-2:active {
  box-shadow: 0 1px 6px #879ba6 inset;
}

.emotion-2:disabled {
  cursor: not-allowed;
  background: #eaeff4;
  color: rgba(135,155,166,0.5);
  box-shadow: none;
}

.emotion-1 {
  margin-left: 8px;
  height: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 {
  color: inherit;
}

<button
  className="emotion-2"
>
  Some content
  <div
    className="emotion-1"
  >
    <i
      className="emotion-0 Drycon Drycon-check"
    />
  </div>
</button>
`;

exports[`Button matches snapshot when is basic 1`] = `
.emotion-0 {
  background: #eaeff4;
  color: #172b4e;
  border-radius: 5px;
  padding: calc(8px * 1.5) 24px;
  outline: 0;
  border: 0;
  -webkit-transition: all 0.15s ease-in-out;
  transition: all 0.15s ease-in-out;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.emotion-0:hover {
  cursor: pointer;
  background: #d3dce6;
}

.emotion-0:active {
  box-shadow: 0 1px 6px #879ba6 inset;
}

.emotion-0:disabled {
  cursor: not-allowed;
  background: #eaeff4;
  color: rgba(135,155,166,0.5);
  box-shadow: none;
}

<button
  className="emotion-0"
>
  Some content
</button>
`;

exports[`Button matches snapshot when is small 1`] = `
.emotion-0 {
  background: #eaeff4;
  color: #172b4e;
  border-radius: 5px;
  padding: calc(8px * 1.5) 24px;
  outline: 0;
  border: 0;
  -webkit-transition: all 0.15s ease-in-out;
  transition: all 0.15s ease-in-out;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: calc(8px - 1px) 8px;
}

.emotion-0:hover {
  cursor: pointer;
  background: #d3dce6;
}

.emotion-0:active {
  box-shadow: 0 1px 6px #879ba6 inset;
}

.emotion-0:disabled {
  cursor: not-allowed;
  background: #eaeff4;
  color: rgba(135,155,166,0.5);
  box-shadow: none;
}

<button
  className="emotion-0"
>
  Some content
</button>
`;
